src/backend/distributed/planner/distributed_planner.c

```diff
@@ -921,8 +924,8 @@ TryCreateDistributedPlannedStmt(PlannedStmt *localPlan,
  * 3. Logical planner
  */
 DistributedPlan *
-CreateDistributedPlan(uint64 planId, Query *originalQuery, Query *query, ParamListInfo
-					  boundParams, bool hasUnresolvedParams,
+CreateDistributedPlan(uint64 planId, List *previousSubPlanList, Query *originalQuery,
```

Comment:
I'm still curious why can't we avoid adding previousSubPlanList param into `CreateDistributedPlan()`. In other words, why do we get incorrect results if we move the error check below the recursive call made `CreateDistributedPlan()`, as we discussed yesterday.